WcaOnRails/app/models/registration_competition_event.rb

```diff
@@ -6,5 +6,5 @@ class RegistrationCompetitionEvent < ApplicationRecord
   belongs_to :competition_event
   validates :competition_event, presence: true
 
-  has_one :event, through: :competition_event
+  delegate :event, to: :competition_event, allow_nil: true
```

Comment:
Ah I resolved my comment a bit too soon: while this is definitely correct for a basic usage, it's worth noting that removing the `has_one` may prevent us from doing `joins` on `event` in some places of the code.
If we don't do it it's fine, but it's worth keeping in mind!

Reply:
Good point, updated!